lesson_16/api/api_app/src/main/java/com/codedifferently/lesson16/web/MediaItemsController.java

```diff
@@ -27,4 +37,31 @@ public GetMediaItemsResponse getItems() {
     var response = GetMediaItemsResponse.builder().items(responseItems).build();
     return response;
   }
+
+  @GetMapping("/items/:id")
+  public GetMediaItemsResponse getMediaItem(@PathVariable String id) {
+    Set<MediaItem> items = library.search(SearchCriteria.builder().id(id).build());
+
+    if (items.isEmpty()) {
+      throw new ResponseStatusException(HttpStatus.NOT_FOUND, "Media item not found");
+    } else {
+      List<MediaItemResponse> responseItems = items.stream().map(MediaItemResponse::from).toList();
+      var response = GetMediaItemsResponse.builder().items(responseItems).build();
+      return response;
+    }
+  }
+
+  @DeleteMapping("/items/{id}")
+  public ResponseEntity<Void> deleteItem(@PathVariable("id") UUID id) {
+    library.removeMediaItem(id, librarian);
+    return ResponseEntity.noContent().build();
+  }
+
+  @PostMapping("/items/")
```

Comment:
You might try removing the trailing forward slash and see if you get a different result.